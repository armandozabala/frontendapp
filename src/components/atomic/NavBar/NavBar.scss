@import "../../../styles/main.scss";

.meli-container {
  max-width: 1080px;
  margin-left: auto;
  margin-right: auto;
  box-sizing: border-box;
  padding-left: $base-container-padding;
  padding-right: $base-container-padding;
}

.meli-search-navbar {
  background-color: $yellow-meli;
  padding: 10px 0;
}

.meli-search-navbar {
  background-color: $yellow-meli;
  padding: $search-navbar-padding 0;
}

.meli-search-navbar-form {
  display: flex;
}

.meli-search-input {
  border-top-left-radius: $search-input-radius;
  border-bottom-left-radius: $search-input-radius;
  font-size: $search-input-font-size;
  border: none;
  width: 100%;
  padding: calc($search-input-padding / 2) $search-input-padding;
  color: $medium-gray;
  flex: 1;
  box-sizing: border-box;

  &::placeholder {
    color: $medium-gray;
  }

  &:focus {
    outline: none;
    box-shadow: none;
  }
}

.meli-search-button {
  padding: 0;
  cursor: pointer;
  box-sizing: border-box;
  border-top-right-radius: $search-input-radius;
  border-bottom-right-radius: $search-input-radius;
  border: none;
  background: url(../../../assets/images/ic_Search.png) $light-gray center
    center no-repeat;
  width: $search-button-size;
  height: $search-button-size;

  &:hover {
    background-color: $search-button-background-hover;
  }
}
