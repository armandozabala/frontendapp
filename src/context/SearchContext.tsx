import { createContext } from "react";

export type SearchContextType = {
  search: string;
  setSearch: Function;
};

export const SearchContext = createContext<SearchContextType>();
