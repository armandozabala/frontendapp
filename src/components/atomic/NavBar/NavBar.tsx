import React, { useContext, useEffect, useState } from "react";
import "./NavBar.scss";
import logo from "@assets/images/Logo_ML.png";

import { useNavigate } from "react-router-dom";
import { SearchContext } from "../../../context/searchContext";

const NavBar = ({ query }) => {
  console.log("object", query);
  const navigate = useNavigate();
  const [inputValue, setInputValue] = useState(query);
  const { search, setSearch } = useContext(SearchContext);

  useEffect(() => {
    setInputValue(search);
  }, [search]);

  useEffect(() => {
    setInputValue(query);
  }, [query]);

  const handleInputChange = (event: any) => {
    setInputValue(event.target.value);
    setSearch(event.target.value);
  };

  const handleSubmit = (e: any) => {
    e.preventDefault();
    if (search.trim().length > 2) {
      navigate(`/items?search=${search}`);
    }
  };
  return (
    <>
      <header className="meli-search-navbar">
        <div className="meli-container">
          <div className="meli-grid meli-flex-middle">
            <div className="meli-width-auto">
              <a className="meli-search-navbar-logo" href="/">
                <img src={logo} alt="Mercadolibre" width="56" height="36" />
              </a>
            </div>

            <div className="meli-width-expand">
              <form className="meli-search-navbar-form" onSubmit={handleSubmit}>
                <input
                  name="input-search"
                  type="text"
                  value={inputValue}
                  onChange={handleInputChange}
                  placeholder="Nunca dejes de buscar"
                  className="meli-search-input"
                />
                <button className="meli-search-button" type="submit"></button>
              </form>
            </div>
          </div>
        </div>
      </header>
    </>
  );
};

export default NavBar;
