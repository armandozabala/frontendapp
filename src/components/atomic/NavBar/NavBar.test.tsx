import React from "react";
import { mount } from "enzyme";
import { SearchContext } from "../../../context/SearchContext";
import NavBar from "./NavBar";
import { MemoryRouter, Router } from "react-router-dom";

describe("NavBar", () => {
  const wrapper = mount(
    <MemoryRouter>
      <SearchContext.Provider
        value={{ search: "Camisa", setSearch: jest.fn() }}
      >
        <NavBar />
      </SearchContext.Provider>
    </MemoryRouter>
  );
  test("debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("simulate input", () => {
    const input = wrapper.find(".meli-search-input").at(0);
    input.simulate("change", {
      target: { value: "Ropa" },
    });

    expect(wrapper.find(".meli-search-input").at(0).props().value).toBe("Ropa");
  });

  test("simulate button", () => {
    const wrapper = mount(
      <MemoryRouter>
        <SearchContext.Provider
          value={{ search: "Camisa", setSearch: jest.fn() }}
        >
          <NavBar />
        </SearchContext.Provider>
      </MemoryRouter>
    );
    const input = wrapper.find(".meli-search-input").at(0);
    input.simulate("change", {
      target: { value: "Camisa" },
    });
    wrapper.find("form").simulate("submit", { preventDefault() {} });
  });

  test("simulate button", () => {
    const wrapper = mount(
      <MemoryRouter>
        <SearchContext.Provider value={{ search: "", setSearch: jest.fn() }}>
          <NavBar />
        </SearchContext.Provider>
      </MemoryRouter>
    );
    const input = wrapper.find(".meli-search-input").at(0);
    input.simulate("change", {
      target: { value: "" },
    });
    wrapper.find("form").simulate("submit", { preventDefault() {} });
  });
});
