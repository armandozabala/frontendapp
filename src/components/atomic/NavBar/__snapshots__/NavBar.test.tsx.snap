// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar debe mostrar el componente correctamente 1`] = `ReactWrapper {}`;
